### ISSUE TITLE
Seeking on a livestream at prepare time

### ISSUE THREAD

**Issue by Jukurrpa** [google/ExoPlayer#1268]
We're implementing playback resuming on an app that plays both live and finished streams (in Mpeg-DASH format).

We do the standard initialize wrapper, and call `prepare()`. Then we wait for it to to report the `STATE_PREPARING` state and call `ExoPlayer.seekTo()` method.

This works well on finished streams with a `seekTo()` parameter in milliseconds relative from the beginning of the file, but not in live streams with an absolute position. It works if we call `seekTo()` upon receiving the `STATE_READY` state instead, but it's obviously not great as it causes a twice as long loading time and the user seeing a frame of the stream at its current head position.

Just in case, the position we attempt to seek to is well under the stream's time shift buffer depth.

**Comment by ojw28** [comment]
Yes, we should do better at this. Although it is somewhat unusual to have this requirement vs. resuming to the live edge, which is what ExoPlayer does currently. Could you clarify the use case where what you're doing makes more sense, for my education? Thanks!

p.s. As a temporary workaround to avoid seeing a frame at the current head position, you can always just put a fully black view over the top of the Surface until you've got the player where you want it and into the ready state.

**Comment by Jukurrpa** [comment]
If the user was watching a live stream and left for whatever reason, they have the possibility to resume the stream right from where they stopped. Of course they can already do it by starting the stream and then seeking on it, so that's mainly for the convenience of resuming in one tap (and one loading).

I guessed this was a pretty rare use case for the player, so glad to ear you guys are aware of it! Keeping the shutter closed will indeed be a partial workaround in the meantime.

**Comment by ojw28** [comment]
I think this is most likely fixed in V2. Please give it a try and see what you discover. Thanks!
